pkg/sql/executor.go

```diff
@@ -446,15 +446,21 @@ func (e *Executor) Prepare(
 
 	// Prepare needs a transaction because it needs to retrieve db/table
 	// descriptors for type checking.
-	// TODO(andrei): is this OK? If we're preparing as part of a SQL txn, how do
-	// we check that they're reading descriptors consistent with the txn in which
-	// they'll be used?
-	txn := client.NewTxn(e.cfg.DB)
-	if err := txn.SetIsolation(session.DefaultIsolationLevel); err != nil {
-		panic(err)
+	txn := session.TxnState.txn
+	if txn == nil {
+		// The new txn need not be the same transaction used by statements following
```

Comment:
I agree we have https://github.com/cockroachdb/cockroach/issues/6774 . Feel free to add more of your concerns to that issue and let's not block this PR.

The reason why the entire leasing mechanism takes an external  transaction is only for the purposes of supporting INSERT after CREATE TABLE.

See this comment: 
https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/lease.go#L155